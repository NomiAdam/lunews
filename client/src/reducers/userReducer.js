import { evolve, always } from 'ramda';
import { createReducer } from '../context/util';

export const userActionTypes = {
    loginUser: 'user/LOGIN_USER',
    logoutUser: 'user/LOGOUT_USER',
    setOffline: 'user/SET_OFFLINE',
    setDarkMode: 'user/SET_DARK_MODE',
    setDownloadSize: 'user/SET_DOWNLOAD',
};

export default DEFAULT_STATE => createReducer(DEFAULT_STATE, {
    [ userActionTypes.loginUser ]: (state, { payload }) => ({
        ...state,
        user: payload,
    }),
    [ userActionTypes.logoutUser ]: state => evolve({ user: always(DEFAULT_STATE.user) }, state),
    [ userActionTypes.setOffline ]: (state, { payload }) => ({
        ...state,
        config: evolve({ offline: always(payload) }, state.config),
    }),
    [ userActionTypes.setDownloadSize ]: (state, { payload }) => ({
        ...state,
        config: evolve({ downloadSize: always(payload) }, state.config),
    }),
    [ userActionTypes.setDarkMode ]: (state, { payload }) => ({
        ...state,
        config: evolve({ darkMode: always(payload) }, state.config),
    }),
});
