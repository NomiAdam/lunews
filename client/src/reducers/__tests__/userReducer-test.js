/* eslint-disable no-undef */
import { logoutUser, setUser, setOffline } from '../userReducer';

const mockPayload = 'mock';

describe('userReducer', () => {
    describe('userReducerActionCreators', () => {
        it('should match the snapshot for setUser', () => {
            expect(setUser(mockPayload)).toMatchSnapshot();
        });
        it('should match the snapshot for logoutUser', () => {
            expect(logoutUser).toMatchSnapshot();
        });
        it('should match the snapshot for setOffline', () => {
            expect(setOffline).toMatchSnapshot();
        });
    });
});
