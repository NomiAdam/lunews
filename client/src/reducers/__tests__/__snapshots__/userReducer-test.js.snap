// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer userReducerActionCreators should match the snapshot for logoutUser 1`] = `
Object {
  "type": "user/LOGOUT_USER",
}
`;

exports[`userReducer userReducerActionCreators should match the snapshot for setOffline 1`] = `[Function]`;

exports[`userReducer userReducerActionCreators should match the snapshot for setUser 1`] = `
Object {
  "payload": "mock",
  "type": "user/LOGIN_USER",
}
`;
